pkg/ipamd/ipamd.go

```diff
@@ -2188,3 +2221,29 @@ func (c *IPAMContext) isConfigValid() bool {
 
 	return true
 }
+
+func (c *IPAMContext) BroadcastMissingPermissionsEvent(nodeInitErr error) {
+	log.Errorf("Failed to start aws-node due to missing policy, verify AmazonEKS_CNI_Policy(or equivalent) is attached to supplied IAM Role, err:%v", nodeInitErr)
+
+	labelSelector := labels.SelectorFromSet(labels.Set{labelK8sapp: awsNode})
+	fieldSelector := fields.SelectorFromSet(fields.Set{specNodeName: c.myNodeName})
+	ctx := context.TODO()
+	listOptions := client.ListOptions{
+		LabelSelector: labelSelector,
+		FieldSelector: fieldSelector,
+	}
+
+	var podList corev1.PodList
+	err := c.rawK8SClient.List(ctx, &podList, &listOptions)
+	if err != nil {
+		log.Errorf("Failed to get pods, cannot broadcast missing permissions event: %v", err)
+		return
+	}
+
+	for _, pod := range podList.Items {
```

Comment:
Will this not be just one pod? Since only one instance of aws-node will be running. Am I missing anything?

Reply:
Yes, in this case, it will be just one pod for the specified label & fields. 
Using the List operation since we set list options. This is also more generic than getting one pod, as it can be used in case we want to raise events on other pods.

Reply:
Yeah but we won't be raising events for other pods. Lets just keep it to aws-node pod.

Reply:
I mean we can't either. We will have no idea why other pods failed.

Reply:
Ok, why do we even need this though? All we need is the Pod name where this blob of code is running. Why can't we just source it via env variable and be done with it. What am I missing? We do that for Node Name right now - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/config/master/aws-k8s-cni.yaml#L207 and we can do the same for Pod Name. Why do we need to make an API Server call here? We can do away with the code where we are creating kubeclient etc.

I'm suggesting something like below - 

```
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
```

Reply:
Thanks! Previously we couldn't get the pod name without listing the pods, but adding the ENV will work.
Will update the PR.

Reply:
We are not saving any API calls by adding ENV for POD_NAME since we still need to make an API call to get the pod object. 
This code is moved to event recorder utils, but the logic will remain same.